backend/api/src/create-market.ts

```diff
@@ -175,7 +181,8 @@ export async function createMarketHelper(body: Body, auth: AuthedUser) {
 
     const contract = getNewContract(
       removeUndefinedProps({
-        id: randomString(),
+        // Risk: idempotencyKey comes from the client, mischievous users could pass unexpected strings.
+        id: idempotencyKey ?? randomString(),
```

Comment:
This allows too much mischief for me! You could just have an in-memory map on the api so we don't have to enter any random strings into the db. It would be reset on api redeploys but this doesn't happen that often, and unlikely to happen between market submissions. Alternatively maybe the zod library can do more checks on the string to make sure it's safe, but I'm not sure.

Reply:
I like the idea of using zod, maybe to do a simple regex validation. I can amend the PR with that if you're interested.